web/app.js

```diff
@@ -940,6 +940,7 @@ const PDFViewerApplication = {
 
     promises.push(this.pdfLoadingTask.destroy());
     this.pdfLoadingTask = null;
+    this.l10n?.pause();
```

Comment:
Unfortunately this will break the GENERIC viewer, since that one supports opening more than one PDF document.

Hence in this *particular case*, since it's a special test-only situation, it's probably best to place this code in to prevent further issues:
https://github.com/mozilla/pdf.js/blob/2f462a8afe9e2d03d0ae376ed4ac0696c36a19b3/web/app.js#L2107-L2113

Reply:
While I do agree that we need to keep the generic viewer use case working, are we sure that this warning can't also occur outside of the tests? I don't think we're really doing much special in the integration tests since it's basically just an "open viewer with PDF, perform actions, close viewer" sequence, so couldn't a "regular" viewer close also trigger this?

Reply:
> While I do agree that we need to keep the generic viewer use case working, are we sure that this warning can't also occur outside of the tests?

Well, we've never seen any indication of that being a problem in practice...

> I don't think we're really doing much special in the integration tests since it's basically just an "open viewer with PDF, perform actions, close viewer" sequence, so couldn't a "regular" viewer close also trigger this?

Sure, but I do believe that there's one difference here: The tests are doing all of those steps in much quicker succession than a normal user would ever be able to.

Reply:
Yes, I think that's fair. Let's indeed limit this to the testing mode then per your suggestion, since that'd resolve the issue at hand and we can always revisit it later should we ever get an indication of this also happening elsewhere.